scrapy/core/scheduler.py

```diff
@@ -195,23 +217,22 @@ def __init__(
         self.logunser: bool = logunser
         self.stats: StatsCollector | None = stats
         self.crawler: Crawler | None = crawler
-
-    @classmethod
-    def from_crawler(cls, crawler: Crawler) -> Self:
-        """
-        Factory method, initializes the scheduler with arguments taken from the crawl settings
-        """
-        dupefilter_cls = load_object(crawler.settings["DUPEFILTER_CLASS"])
-        return cls(
-            dupefilter=build_from_crawler(dupefilter_cls, crawler),
-            jobdir=job_dir(crawler.settings),
-            dqclass=load_object(crawler.settings["SCHEDULER_DISK_QUEUE"]),
-            mqclass=load_object(crawler.settings["SCHEDULER_MEMORY_QUEUE"]),
-            logunser=crawler.settings.getbool("SCHEDULER_DEBUG"),
-            stats=crawler.stats,
-            pqclass=load_object(crawler.settings["SCHEDULER_PRIORITY_QUEUE"]),
-            crawler=crawler,
+        self._sdqclass: type[BaseQueue] | None = self._get_start_queue_cls(
+            crawler, "DISK"
         )
+        self._smqclass: type[BaseQueue] | None = self._get_start_queue_cls(
+            crawler, "MEMORY"
+        )
```

Comment:
Is it done here (instead of passing instances to `__init__`) to simplify backwards compatibility handling?

Reply:
Yes. I kind of default to that nowadays, specially for things that can be overridden through a setting. But it may make sense to allow subclasses to override these, maybe ignore the setting or use some other setting to decide.

@wRAR Shall I make them public?

If in doubt, we can discuss it with @kmike for a follow-up release.

Reply:
I'm fine with either approach. My main concern is about consistency (some things are passed to `__init__`, others are created later), but it's not a big deal, if it helps to make a backwards compatible change with less code to maintain.

Reply:
I'm fine with keeping the current code, it's not the first time we do newer things differently from the older ones.